main.rb

```diff
@@ -0,0 +1,33 @@
+def bubble_sort(array)
+  n = array.length
+  loop do
+    swap = false
+    (n - 1).times do |i|
+      if array[i] > array[i + 1]
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swap = true
+      end
+    end
+    break unless swap
+  end
+  array
+end
+arr = [1, 3, 4, 2, 6, 5]
+print bubble_sort(arr)
+
+def bubble_sort_by(array)
+  n = array.length
+  loop do
+    swap = false
+    (n - 1).times do |i|
+      if yield(array[i], array[i + 1])
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swap = true
+      end
+    end
+    break unless swap
+  end
+  array
+end
+arr = [1, 5, 3, 6, 2, 4, 1]
+print bubble_sort_by(arr) { |arg1, arg2| arg1.length > arg2.length }
```

Comment:
- [ ] your input(arr) on line 32 is causing your code to break. The bubble_sort_by method is not for testing an array of numbers. It breaks on line 32 because at that point you when the yield is called on array[i] and array[i + 1], your code will try to do this `1.length > 5.length` which is wrong. So all you need to do is change your arg for that bubble_sort_by method. It should be used to test such input like `['hey', 'hello', 'school']`. I hope you get the point.